app/controllers/users_controller.rb

```diff
@@ -36,12 +37,29 @@ def update
     end
   end
 
+  def set_admin
+    @user = User.find(params[:id])
+    if @user.admin?
+      @user.update_attribute(:admin, 0)
+    else
+      @user.update_attribute(:admin, 1)
+    end
+
+    redirect_to users_url
+  end
+
   private
     def correct_user
       @user = User.find(params[:id])
       redirect_to(root_url) unless current_user?(@user)
     end
 
+    def admin_user
+      unless current_user.admin?
+        flash.now[:danger] = "You are not authorized."
```

Comment:
@CharlotteDS
- This filter function doesn't stop the `set_admin` request if user is not admin, and flas message is not displayed as well.